### PR TITLE
[stable/prometheus-rabbitmq-exporter] Adding Prometheus Operator Service Monitor

### PR DESCRIPTION
**Pull request by RaymondArias**
#### What this PR does / why we need it:
Adds a Service Monitor resource to allow for easier integration with Prometheus Operator

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/prometheus-rabbitmq-exporter/templates/monitor.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  {{- if .Values.prometheus.monitor.additionalLabels }}
+{{ toYaml .Values.prometheus.monitor.additionalLabels | indent 4 }}
+  {{- end }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "prometheus-rabbitmq-exporter.name" . }}
+      release: {{ .Release.Name }}
+  endpoints:
+    - port: rabbitmq-exporter
```
Comment (steven-sheehy) [inline stable/prometheus-rabbitmq-exporter/templates/monitor.yaml]: Named port should be 'publish' to match deployment, shouldn't it? It's not scraping the service.

File: stable/prometheus-rabbitmq-exporter/README.md [hunk 2]
```diff
@@ -65,6 +65,8 @@ The following table lists the configurable parameters and their default values.
 | `rabbitmq.timeout`       | timeout in seconds for retrieving data from management plugin          | `30`                      |
 | `rabbitmq.max_queues`    | max number of queues before we drop metrics (disabled if set to 0)     | `0`                       |
 | `annotations`            | pod annotations for easier discovery                                   | {}                        |
+| `prometheus.monitor.enabled` | Set this to `true` to create ServiceMonitor for Prometheus operator | `false` | |
+| `prometheus.monitor.additionalLabels` | Additional labels that can be used so ServiceMonitor will be discovered by Prometheus | `{}` | |
```
Comment (steven-sheehy) [inline stable/prometheus-rabbitmq-exporter/README.md]: A `prometheus.monitor.interval` option would be nice

File: stable/prometheus-rabbitmq-exporter/templates/monitor.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    chart: {{ template "prometheus-rabbitmq-exporter.chart" . }}
+    heritage: {{ .Release.Service }}
+  {{- if .Values.prometheus.monitor.additionalLabels }}
+{{ toYaml .Values.prometheus.monitor.additionalLabels | indent 4 }}
+  {{- end }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "prometheus-rabbitmq-exporter.name" . }}
+      release: {{ .Release.Name }}
```
Comment (steven-sheehy) [inline stable/prometheus-rabbitmq-exporter/templates/monitor.yaml:17]: Should probably have a namespaceSelector to narrow down to release namespace

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: stable/prometheus-rabbitmq-exporter/templates/monitor.yaml (hunk 1 above)
Comment (RaymondArias) [reply stable/prometheus-rabbitmq-exporter/templates/monitor.yaml]: Port name changed to publish

File: stable/prometheus-rabbitmq-exporter/README.md (hunk 2 above)
Comment (RaymondArias) [reply stable/prometheus-rabbitmq-exporter/README.md]: Added interval to servicemonitor

File: stable/prometheus-rabbitmq-exporter/templates/monitor.yaml (hunk 3 above)
Comment (RaymondArias) [reply stable/prometheus-rabbitmq-exporter/templates/monitor.yaml:17]: added namespace to servicemonitor

**Comment by steven-sheehy** [comment]
Need to sign DCO

**Comment by steven-sheehy** [comment]
/approve
Please assign to @viglesiasce and @mattfarina for lgtm

**Comment by mattfarina** [comment]
cc @juanchimienti

**Comment by RaymondArias** [comment]
@mattfarina or @viglesiasce can I get either one of you guys to review this?

**Review by mattfarina — APPROVED** [review]

**Pull request merged**